.template-helpers/variant.sh

```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
+set -e
+
+repo="$1"
+if [ -z "$repo" ]; then
+	echo >&2 "usage: $0 repo"
+	echo >&2 "   ie: $0 hylang"
+	exit 1
+fi
+
+dir="$(dirname "$(readlink -f "$BASH_SOURCE")")"
+
+IFS=$'\n'
+lines=( $(curl -sSL 'https://raw.githubusercontent.com/docker-library/official-images/master/library/'"$repo" | grep -vE '^$|^#') )
+unset IFS
+
+tags=()
+# this loop finds all the "variants", making assumptions about what they can contain
+for line in "${lines[@]}"; do
+	tag="$(echo "$line" | awk -F ': +' '{ print $1 }' | sed -r 's/[0-9.-]+//g')"
```

Comment:
I think that is a case of premature optimization by me to cut things from the later loop.  I think I'll just try the piping and skip this loop entirely (technically it will live on in the pipes).